app/components/trainees/confirmation/contact_details/view.rb

```diff
@@ -1,15 +1,29 @@
 module Confirmation
   class Trainees::Confirmation::ContactDetails::View < GovukComponent::Base
-    attr_accessor :trainee
+    include SanitizeHelper
 
-    delegate :phone_number, :email, to: :trainee
+    attr_accessor :trainee
 
     def initialize(trainee:)
       @trainee = trainee
     end
 
     def address
-      uk_locale? ? uk_address : international_address
+      return I18n.t("components.confirmation.not_provided") if trainee.locale_code.nil?
+
+      sanitize(uk_locale? ? uk_address : international_address)
+    end
+
+    def email
+      return I18n.t("components.confirmation.not_provided") if trainee.email.nil?
```

Comment:
I think this can just be:

```suggestion
return I18n.t("components.confirmation.not_provided") unless trainee.email
```